### PR TITLE
Update Artman grpc tasks for Java

### PR DESCRIPTION
**Pull request by shinfan**
- Rename the output package from `proto-*` to `grpc-*` (Java gradle uses the folder name as the default artifact ID)
- Users may have different versions of protoc plugins installed. Do **NOT** copy the generated java classes from Artman, using the protoc gradle plugin instead (fetching versioned protoc from maven).

### CODE REVIEW

File: test/testdata/python_grpc_client_nopub_pipeline.baseline [hunk 1]
```diff
@@ -1,4 +1,4 @@
-mkdir -p {OUTPUT}
-protoc --proto_path=test/fake-repos/gapi-core-proto/src/main/proto/ --proto_path=test/fake-repos/fake-mixin-proto --proto_path=test/fake-repos/fake-proto --proto_path=MOCK_GRADLE_TASK_OUTPUT --include_imports --include_source_info -o {OUTPUT}/google-cloud-library-v1.desc test/fake-repos/fake-proto/fake.proto test/fake-repos/fake-mixin-proto/fake_mixin.proto
-mkdir -p {OUTPUT}/google-cloud-library-v1-gen-python
-test/fake-repos/toolkit/gradlew -p test/fake-repos/toolkit runGrpcMetadataGen -Pclargs=--descriptor_set={OUTPUT}/google-cloud-library-v1.desc,--input={OUTPUT}/google-cloud-library-v1-gen-python,--output={OUTPUT}/python/proto-google-cloud-library-v1,--metadata_config={CWD}/test/testdata/test_output/google-cloud-library-v1_package.yaml,--language=python,--service_yaml={CWD}/test/testdata/gapi-example-library-proto/src/main/proto/google/example/library/library.yaml
+ mkdir -p {CWD}/test/testdata/test_output
```
Comment (geigerj) [inline test/testdata/python_grpc_client_nopub_pipeline.baseline]: the new leading space is weird...

File: pipeline/tasks/package_metadata_tasks.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def execute(self, api_name, api_version, organization_name, toolkit_path,
                 language):
         api_full_name = task_utils.api_full_name(
             api_name, api_version, organization_name)
-        pkg_dir = os.path.join(output_dir, language, 'proto-' + api_full_name)
+        pkg_dir = os.path.join(output_dir, language, 'grpc-' + api_full_name)
```
Comment (geigerj) [inline pipeline/tasks/package_metadata_tasks.py]: Python is going to want `proto` here. Can we select the value by language?

File: test/testdata/python_grpc_client_nopub_pipeline.baseline (hunk 1 above)
Comment (shinfan) [inline test/testdata/python_grpc_client_nopub_pipeline.baseline]: That was a typo. Fixed.

File: pipeline/tasks/package_metadata_tasks.py (hunk 2 above)
Comment (shinfan) [inline pipeline/tasks/package_metadata_tasks.py]: Done

File: test/testdata/python_grpc_client_nopub_pipeline.baseline [hunk 3]
```diff
@@ -1,4 +1,4 @@
-mkdir -p {OUTPUT}
-protoc --proto_path=test/fake-repos/gapi-core-proto/src/main/proto/ --proto_path=test/fake-repos/fake-mixin-proto --proto_path=test/fake-repos/fake-proto --proto_path=MOCK_GRADLE_TASK_OUTPUT --include_imports --include_source_info -o {OUTPUT}/google-cloud-library-v1.desc test/fake-repos/fake-proto/fake.proto test/fake-repos/fake-mixin-proto/fake_mixin.proto
-mkdir -p {OUTPUT}/google-cloud-library-v1-gen-python
-test/fake-repos/toolkit/gradlew -p test/fake-repos/toolkit runGrpcMetadataGen -Pclargs=--descriptor_set={OUTPUT}/google-cloud-library-v1.desc,--input={OUTPUT}/google-cloud-library-v1-gen-python,--output={OUTPUT}/python/proto-google-cloud-library-v1,--metadata_config={CWD}/test/testdata/test_output/google-cloud-library-v1_package.yaml,--language=python,--service_yaml={CWD}/test/testdata/gapi-example-library-proto/src/main/proto/google/example/library/library.yaml
+mkdir -p {CWD}/test/testdata/test_output
```
Comment (geigerj) [inline test/testdata/python_grpc_client_nopub_pipeline.baseline]: Out of a curiosity, what is causing this change?

Comment (shinfan) [inline test/testdata/python_grpc_client_nopub_pipeline.baseline]: I think those two are equivalent... I will just revert these two python baselines

Comment (shinfan) [inline test/testdata/python_grpc_client_nopub_pipeline.baseline]: Done

**Pull request merged**